### PR TITLE
Populate __name__ for torch.nn.modules.utils.{_single,_pair,...}

### PR DESCRIPTION
**Pull request by jansel**
This helps with debug printouts and python level graph analysis.

### CODE REVIEW

**Review by albanD — COMMENTED** [review]
Isn't functools wraps https://docs.python.org/3/library/functools.html#functools.wraps a better thing to use here?

**Comment by jansel** [comment]
@albanD what would the "wrapped" function be that wraps requires?  As written this isn't wrapping an existing function, but creating a new one.

**Comment by albanD** [comment]
In this case, "parse" is kind of wrapping the different functions "_single", etc.

What about something simpler like this then?
```python
def _ntuple(x, n):
    if isinstance(x, collections.abc.Iterable):
        return tuple(x)
    return tuple(repeat(x, n))

def _single(x):
    return _ntuple(x, 1)
def _pair(x):
    return _ntuple(x, 2)
def _triple(x):
    return _ntuple(x, 3)
def _quadruple(x):
    return _ntuple(x, 4)
```

**Comment by jansel** [comment]
@albanD that would work, however:
1) It would be slightly slower as it introduces an extra call to the call stack.
2) It would break other uses of `_ntuple`.  There are a few in padding.py for `n=6`, and there may be some outside the project as a result of people copying code from padding.py.

I prefer it this way, however if you strongly prefer the rewrite I'll change it.

**Review by albanD — APPROVED** [review]
Following some offline discussion, this seems to be the simplest solution and an ok thing to do.
It is pretty harmless in any case.
